feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go

```diff
@@ -302,8 +302,12 @@ func testRecursiveIPv4Entry(t *testing.T, args *testArgs) {
 			t.Errorf("next-hop index is incorrect: got %v, want %v", got, want)
 		}
 		nh := gnmi.Get(t, args.dut, gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance).Afts().NextHop(nhIndexInst).State())
+		// for devices that return  the nexthop with resolving it recursively, e.g., for a->b->c the device returns c
 		if got, want := nh.GetIpAddress(), atePort2.IPv4; got != want {
-			t.Errorf("next-hop is incorrect: got %v, want %v", got, want)
+			// for devices that return the nexthop without resolving it recursively e.g., for a->b->c the device returns b
+			if got, want := nh.GetIpAddress(), ateIndirectNH; got != want {
```

Comment:
Seems like it only validates that a->b, but does not validate b->c?  

Shall we use this PR (or at least add a TODO here) to deliver some holistic/generic solution?  Maybe a recursive helper function? something like (pseudo code) the following, which should be handle whether a vendor OS returns the flatten next hop or not, right? 
```
func recursiveNextHop(destIP, aftEntries) nextHop {
	nextHopIP, isFound := aftEntries.nexthop(destIP)
	if !isFound {
		return destIP
	}
	return recursiveNextHop(nextHopIP, aftEntries)
}
```

Reply:
the check for b->c is done after the check for a->b https://github.com/openconfig/featureprofiles/blob/5890cc99fb3eec7a4738a5bf264611ee9fb00daa/feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go#L317.

concerning the helper, I agree with that we need a proper helper function here. Will add TODO for now.

Reply:
a TODO sounds good to me. 

can you help add a comment here saying that b->c is validated below so that other readers later doesn't have the same question? Thanks.

Reply:
done